### PR TITLE
Issue #206: SuppressionPatchXpathFilter: Regexp's context strategy

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #206: SuppressionPatchXpathFilter: Regexp's context strategy

this problem is that the all violations' information line is `0`, similar to Checks: `UniqueProperties`, `RegexpMultiline`.

common pattern https://github.com/checkstyle/patch-filters/issues/8#issuecomment-667101466

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/Regexp/newline/defaultContextConfig.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<module name="Checker">
+    <module name="TreeWalker">
+        <module name="Regexp">
+            <property name="format" value="// This code is copyrighted\n// \(c\) MyCompany"/>
+        </module>
+
+        <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+            <property name="file" value="${tp}/defaultContext.patch" />
+            <property name="strategy" value="newline" />
+            <property name="neverSuppressedChecks" value="RegexpCheck" />
```
Comment (rdiachenko) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/Regexp/newline/defaultContextConfig.xml:14]: Why do you need this property for a newline strategy?

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/Regexp/patchedline/defaultContextConfig.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<module name="Checker">
+    <module name="TreeWalker">
+        <module name="Regexp">
+            <property name="format" value="// This code is copyrighted\n// \(c\) MyCompany"/>
+        </module>
+
+        <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+            <property name="file" value="${tp}/defaultContext.patch" />
+            <property name="strategy" value="patchedline" />
+            <property name="neverSuppressedChecks" value="RegexpCheck" />
```
Comment (rdiachenko) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/Regexp/patchedline/defaultContextConfig.xml:14]: Why do you need this property for a patchedline strategy?

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/Regexp/newline/defaultContextConfig.xml (hunk 1 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/Regexp/newline/defaultContextConfig.xml:14]: Removed, as mentioned in https://github.com/checkstyle/patch-filters/pull/207#issue-460111691, this check is special, its violation line is `0`, so I added it at the beginning. And yes this situation also should be deal with in `context`.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/Regexp/patchedline/defaultContextConfig.xml (hunk 2 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionpatchxpathfilter/Regexp/patchedline/defaultContextConfig.xml:14]: same answer https://github.com/checkstyle/patch-filters/pull/207#discussion_r467540648

**Pull request merged**